mvlearn/embed/kcca.py

```diff
@@ -201,6 +210,12 @@ def __init__(
         if self.constant < 0 or not (type(self.constant) == float
                                      or type(self.constant) == int):
             raise ValueError("constant must be a positive integer")
+        if self.decomp == "icd":
+            if self.mrank < 0 or not (type(self.mrank) == int):
+                raise ValueError("mrank must be a positive integer no greater \
+                                 than the number of components")
+            if self.precision < 0 or not type(self.precision) == float:
+                raise ValueError("precision must be a positive float")
 
```

Comment:
Need to check to verify that self.method == 'keterring-like', otherwise if they put something else in, it won't do some steps